### ISSUE TITLE
Unable to verify JWT with delegate signer

### ISSUE THREAD

**Issue by m-yahya** [decentralized-identity/did-jwt#55]
**Describe the bug**
`verifyJWT` can't verify a `JWT` signed by a delegate signer.
It gives the error:
`UnhandledPromiseRejectionWarning: Error: Signature invalid for JWT`

**Versions**
`"did-jwt": "^2.0.0"`
 `"did-resolver": "^1.0.0"`
`"ethr-did": "^1.1.0"`
`"ethr-did-resolver": "^1.0.1"`
`"web3": "^1.2.1"`

**Sample Code**
```
let test = async () => {

    const ethrDid = new EthrDID({ provider, registry, address });
    let did = ethrDid.did

    // add delegate
    // const delegate = await ethrDid.addDelegate('0xBa3E28a5fFF2b7e1a6A6D28d8Ed352d2521Af966', {
    //     expiresIn, delegateType
    // })
    // console.log(delegate);

    const newSigner = SimpleSigner('DELEGATE_PRIVAT_KEY')

    const ethrDidResolver = getResolver({ provider: provider, registry: registry })
    const didResolver = new Resolver(ethrDidResolver)

    let jwt = '';
    createJWT(
        {
            aud: 'did:ethr:0x4b791b0ddadd64cc935381251f605355e9c7676b',
            name: 'uPort Developer',
            age: 23,
            exp: 1957463421
        },
        {
            alg: 'ES256K-R',
            issuer: 'did:ethr:0x7781ddd1af9c3a1fa168fc4511b01cd91f1dfad0',
            signer: newSigner
        }
    ).then(jwt => {
        setTimeout(function () {
            verifyJWT(jwt, { audience: 'did:ethr:0x4b791b0ddadd64cc935381251f605355e9c7676b', auth: false }).then((verifiedRespone) => {
                // gives error
                console.log(verifiedRespone);
            });
        }, 5000);
    });

}

test();
```

**To Reproduce**
1. init npm project
2. paste the above code into `index.js`
3. config the provider
4. replace the `address` and `registry` with your own stuff
5. install the above listed dependencies
6. add a new delegate signer
7. create a newSigner with the private key of new delegate signer
8. use new delegate signer to create JWT
9. run `node index.js`

**Expected behavior**
A verified did-JWT returns an object.

**Additional context**
The same `JWT` is verifiable using the original signer. The `JWT` is also verifiable with downgraded version of `did-jwt@^0.2.0`.

**Comment by mirceanis** [comment]
This is a known issue.
The latest release of did-jwt can only use the default mainnet registry for verification of JWTs issued by ethr-dids.
Please use v0.2.0 until a fix comes out.

We're working on a fix where developers can choose the set of DID resolvers (that they can also configure) they want to work with when verifying JWTs.

cc @mi-xu

**Comment by mirceanis** [comment]
This was fixed in #57 and released in [v4.0.0](https://github.com/decentralized-identity/did-jwt/releases/tag/v4.0.0)
